bin/improver-regrid

```diff
@@ -105,10 +100,15 @@ def main():
             extrapolation_mode=args.extrapolation_mode)
 
     source_on_target = source_data.regrid(target_grid, regridder)
-    target_grid_attributes = {k: v for (k, v) in target_grid.attributes.items()
-                              if 'mosg__grid' in k}
-    amend_metadata(source_on_target, revised_attributes=target_grid_attributes)
-    delete_attributes(source_on_target, ['title'])
+    if args.json_file:
+        metadata_dict = load_json(args.json_file)
+        amend_metadata(source_on_target, **metadata_dict)
+    else:
+        target_grid_attributes = (
```

Comment:
Does this need to happen anyway, even if a json file is included too?

Reply:
Yes, I've moved this ahead of the metadata dict usage.